File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> a = new ArrayList<>();
        for(int i = 0 ; i < arr.length; i++){
            if( i != 0 ){
                if(arr[i] != arr[i-1])
                a.add (arr[i]);
            } else {
                a.add(arr[i]);
            }
        }
        return a.stream().mapToInt( i -> i).toArray();
    }
}

File: JavaScript/프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr){
    let prevNum = -1;
    const answer = [];
    for(const n of arr){
        if(prevNum === -1){
            answer.push(n);
            prevNum = n;
        } else if(prevNum !== n){
            answer.push(n);
            prevNum = n;
        }
    }
    return answer;
}